apps/wallet-mobile/src/legacy/Staking/StakingCenter/StakingCenter.tsx

```diff
@@ -49,10 +57,17 @@ export const StakingCenter = () => {
       onSuccess: (yoroiUnsignedTx) => {
         if (selectedPoolId == null) return
 
-        navigation.navigate('delegation-confirmation', {
-          poolId: selectedPoolId,
-          yoroiUnsignedTx,
+        let operations = [<DelegateStakeOperation poolId={selectedPoolId} key="0" />]
+        if (!hasStakingKeyRegistered) operations = [<RegisterStakingKeyOperation key="-1" />, ...operations]
```

Comment:
I will definitely remove this dependency by matching operations and certificates. I will only leave this feature for operations that are not defined by certificates.